### PR TITLE
ISPN-5710 Unknown entity name when indexing entities defined in inner…

### PR DESCRIPTION
**Pull request by gustavonalle**
… classes

https://issues.jboss.org/browse/ISPN-5710

### CODE REVIEW

File: query/src/test/java/org/infinispan/query/dsl/embedded/SingleClassDSLQueryTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+              .addProperty("default.directory_provider", "ram")
+              .addProperty("lucene_version", "LUCENE_CURRENT");
+      return new DefaultCacheManager(builder.build());
+   }
+
+   @Test
+   public void testQuery() throws Exception {
+      Cache<String, Person> cache = cacheManager.getCache();
+      cache.put("person1", new Person("William", "Shakespeare"));
+      QueryFactory<?> queryFactory = Search.getQueryFactory(cache);
```
Comment (tristantarrant) [inline query/src/test/java/org/infinispan/query/dsl/embedded/SingleClassDSLQueryTest.java]: I've seen this code before :)

Comment (gustavonalle) [inline query/src/test/java/org/infinispan/query/dsl/embedded/SingleClassDSLQueryTest.java]: How rude of me. Updated to give proper credit  :smile:

**Comment by anistor** [comment]
Looking ..

**Comment by danberindei** [comment]
Integrated, thanks Gustavo! And sorry for missing your comment, Adrian :)

**Comment by anistor** [comment]
I see another 3 places where getCanonicalName() should be replaced with getName()

**Comment by anistor** [comment]
New PR here https://github.com/infinispan/infinispan/pull/3681

**Pull request closed without merging**